src/lib/content/coordinators/GetByHierarchy/UrlBuilder.ts

```diff
@@ -0,0 +1,39 @@
+import { HierarchyRequest } from '../../model/ByHierachy';
+import { encodeQueryString } from '../../../utils/Url';
+
+export class HierarchyURLBuilder {
+  static HIERARCHY_URL = '/content/hierarchies/descendants/id/';
+  static MAXIMUM_DEPTH_PARAM = 'hierarchyDepth';
+  static MAXIMUM_PAGE_SIZE_PARAM = 'maxPageSize';
+  static LAST_EVALUATED_PARAM = 'pageCursor';
+
+  buildUrl(request: HierarchyRequest): string {
```

Comment:
should ideally be using URL & URLSearchParams here, aswell as avoiding  != in favour of !==